src/builtin.d

```diff
@@ -48,35 +49,35 @@ extern (C++) Expression eval_sin(Loc loc, FuncDeclaration fd, Expressions* argum
 {
     Expression arg0 = (*arguments)[0];
     assert(arg0.op == TOKfloat64);
-    return new RealExp(loc, sinl(arg0.toReal()), arg0.type);
+    return new RealExp(loc, TargetReal.sin(arg0.toReal()), arg0.type);
```

Comment:
> Port's reason for existence is to remove #if stuff from the front end.

It also implements functionality that may not be available on every platform (``memicmp``/``stricmp``).  I think it is safe to say that long double math functions also falls into this category too, it's just that DMD has never had enough widespread support to run into the problem.